presto-spi/src/main/java/io/prestosql/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -353,77 +352,42 @@ else if (compare < 0) {
      */
     private static long addUnsignedReturnOverflow(Slice left, Slice right, Slice result, boolean resultNegative)
     {
-        // TODO: consider two 7 bytes operations
-        int l0 = getInt(left, 0);
-        int l1 = getInt(left, 1);
-        int l2 = getInt(left, 2);
-        int l3 = getInt(left, 3);
+        long l0 = getLong(left, 0);
+        long l1 = getLong(left, 1);
 
-        int r0 = getInt(right, 0);
-        int r1 = getInt(right, 1);
-        int r2 = getInt(right, 2);
-        int r3 = getInt(right, 3);
+        long r0 = getLong(right, 0);
+        long r1 = getLong(right, 1);
 
-        long intermediateResult;
-        intermediateResult = toUnsignedLong(l0) + toUnsignedLong(r0);
+        long z0 = l0 + r0;
+        // Long.unsignedCompare() inlined
```

Comment:
It's not inlined method as `unsignedCompare` can return `-1, 0 or 1`.

Could you extract `unsignedIsSmaller` method and add a comment what it's based on?